src/SystemWebAdapters/src/HttpResponse.cs

```diff
@@ -56,43 +69,98 @@ public HttpCookieCollection Cookies
 
         public bool SuppressContent
         {
-            get => throw new NotImplementedException();
-            set => throw new NotImplementedException();
+            get => BufferedFeature.SuppressContent;
+            set => BufferedFeature.SuppressContent = value;
+        }
+
+        public Encoding ContentEncoding
+        {
+            get => TypedHeaders.ContentType?.Encoding ?? Encoding.UTF8;
+            set
+            {
+                var contentType = TypedHeaders.ContentType;
+
+                if (contentType is null)
+                {
+                    contentType = new(value.WebName);
+                    TypedHeaders.ContentType = contentType;
+                }
+                else
+                {
+                    contentType.Encoding = value;
```

Comment:
This isn't getting written back to the headers.
```suggestion
                    contentType.Encoding = value;
                    TypedHeaders.ContentType = contentType;
```